augur/refine.py

```diff
@@ -309,6 +309,16 @@ def run(args):
         except InvalidYearBounds as error:
             raise AugurError(f"Invalid value for --year-bounds: {error}")
 
+        terminal_names = {n.name for n in T.get_terminals()}
+        matched_ids = terminal_names & set(dates.keys())
+
+        if len(matched_ids) < len(terminal_names) * 0.5:
```

Comment:
Suggestion: warn on **any** missing metadata. That would match the expectation for typical Augur workflows. `augur filter` drops data on either side if it's missing on the other side.